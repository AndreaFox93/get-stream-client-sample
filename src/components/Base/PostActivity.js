import { get } from 'lodash'
import moment from 'moment'
import PropTypes from 'prop-types'
import React from 'react'
import { Link } from 'react-router-dom'
import Box from '../Ui/Box'
import Text from '../Ui/Text'
import BaseAttachment from './BaseAttachment'
import BasePostContent from './BasePostContent'
import UserLine from './UserLine'

function renderAttachments(attachments) {
  return Object.keys(attachments).map(key => {
    const currentAttachment = attachments[key]

    return(
        <Box key={key} marginTop={10}>
          <BaseAttachment attachment={currentAttachment} />
        </Box>
    )
  })
}

const PostActivity = props => {
  const { activity } = props
  const { activity: { actor, object, time } } = props

  const { name, profileImage } = actor.data

  return (
      <Box display="flex" direction="column" marginBottom={30}>
        <Box marginBottom={15}>
          <Link to={`/users/${actor.id}`}>
            <UserLine username={name} image={profileImage} imageShape="circle" />
          </Link>
          <Box marginTop={4}>
            <Text color="grayBlue" fontSize="sm">{moment(time).format('Do MMMM [at] H:mm')}</Text>
          </Box>
        </Box>
        <Box>
          <BasePostContent content={object} />
        </Box>
        <Box>
          {renderAttachments(get(activity, 'attachments', {}))}
        </Box>
      </Box>
  )
}

PostActivity.propTypes = {
  activity: PropTypes.object.isRequired,
}

export default PostActivity
