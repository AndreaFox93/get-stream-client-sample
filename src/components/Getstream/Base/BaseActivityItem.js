import React from 'react'
import PropTypes from 'prop-types'

const BaseActivityItem = (props) => {
  const { activity } = props

  return (
      <div>
        {JSON.stringify(activity, null, 4)}
      </div>
  )
}

BaseActivityItem.propTypes = {
  activity: PropTypes.object,
}

BaseActivityItem.defaultProps = {
  activity: {},
}

export default BaseActivityItem
