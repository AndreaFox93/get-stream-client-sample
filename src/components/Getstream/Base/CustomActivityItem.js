import React from 'react'
import PropTypes from 'prop-types'

const CustomActivityItem = (props) => {
  const { activity } = props

  return (
      <div>
        {JSON.stringify(activity, null, 4)}
      </div>
  )
}

CustomActivityItem.propTypes = {
  activity: PropTypes.object,
}

CustomActivityItem.defaultProps = {
  activity: {},
}

export default CustomActivityItem
