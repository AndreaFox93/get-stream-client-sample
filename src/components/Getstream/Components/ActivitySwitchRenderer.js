import { get } from 'lodash'
import PropTypes from 'prop-types'
import { safeRender } from '../utils'

type Props = {
  verbs: Object,
  activity: Object
}

const ActivitySwitchRenderer = (props): Props => {
  const { activity: { verb }, verbs } = props

  const component = get(verbs, verb, null)

  return safeRender(component, props)
}

ActivitySwitchRenderer.propTypes = {
  verbs: PropTypes.object,
  activity: PropTypes.object.isRequired,
}

export default ActivitySwitchRenderer
