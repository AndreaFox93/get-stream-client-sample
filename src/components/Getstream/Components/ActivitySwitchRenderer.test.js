import React from 'react'
import renderer from 'react-test-renderer'
import PostActivity from '../../Base/PostActivity'
import BaseActivityItem from '../Base/BaseActivityItem'
import CustomActivityItem from '../Base/CustomActivityItem'
import ActivitySwitchRenderer from './ActivitySwitchRenderer'
import { MemoryRouter } from 'react-router-dom'

const activity = {
  'actor': {
    'id': 'purpletiger922',
    'collection': 'user',
    'data': {
      'name': 'Benjamin hansen',
      'profileImage': 'https://randomuser.me/api/portraits/med/men/97.jpg',
    },
    'created_at': '2018-10-18T12:30:17.759825Z',
    'updated_at': '2018-10-18T12:30:17.759825Z',
  },
  'foreign_id': '',
  'id': '9245acb5-d2d1-11e8-a18f-0a544dd61b80',
  'object': '@beautifulpanda108 let\'s get coffee one these days â˜•',
  'origin': 'user:purpletiger922',
  'target': '',
  'time': '2018-10-18T12:30:17.883769',
  'verb': 'post',
}

it('renders correctly', () => {
  const tree = renderer
      .create(
          <MemoryRouter>
            <ActivitySwitchRenderer
                activity={activity}
                verbs={{
                  post: PostActivity,
                  reply: CustomActivityItem,
                }}
            />
          </MemoryRouter>,
      )
      .toJSON()
  expect(tree).toMatchSnapshot()
})
