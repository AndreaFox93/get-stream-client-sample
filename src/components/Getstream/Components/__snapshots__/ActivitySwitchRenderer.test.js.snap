// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="flex directionColumn justifyInitial square inherit"
  style={
    Object {
      "height": "auto",
      "margin": 0,
      "marginBottom": 30,
      "padding": 0,
      "width": "auto",
      "zIndex": "auto",
    }
  }
>
  <div
    className="block square inherit"
    style={
      Object {
        "height": "auto",
        "margin": 0,
        "marginBottom": 15,
        "padding": 0,
        "width": "auto",
        "zIndex": "auto",
      }
    }
  >
    <a
      href="/users/purpletiger922"
      onClick={[Function]}
    >
      <div
        className="flex directionRow justifyInitial square inherit"
        style={
          Object {
            "height": "auto",
            "margin": 0,
            "padding": 0,
            "width": "auto",
            "zIndex": "auto",
          }
        }
      >
        <div
          className="block square inherit"
          style={
            Object {
              "height": "auto",
              "margin": 0,
              "marginRight": 5,
              "padding": 0,
              "width": "auto",
              "zIndex": "auto",
            }
          }
        >
          <div
            className="block square inherit"
            style={
              Object {
                "height": 25,
                "margin": 0,
                "padding": 0,
                "width": 25,
                "zIndex": "auto",
              }
            }
          >
            <div
              className="block vividGambodgeBg circle inherit"
              style={
                Object {
                  "height": 25,
                  "margin": 0,
                  "padding": 0,
                  "width": 25,
                  "zIndex": "auto",
                }
              }
            >
              <img
                alt=""
                className="fullWidth circle"
                src="https://randomuser.me/api/portraits/med/men/97.jpg"
              />
            </div>
          </div>
        </div>
        <div
          className="flex directionInitial justifyInitial alignMiddle square inherit"
          style={
            Object {
              "height": "auto",
              "margin": 0,
              "padding": 0,
              "width": "auto",
              "zIndex": "auto",
            }
          }
        >
          <span
            className="bold black"
          >
            Benjamin hansen
          </span>
        </div>
      </div>
    </a>
    <div
      className="block square inherit"
      style={
        Object {
          "height": "auto",
          "margin": 0,
          "marginTop": 4,
          "padding": 0,
          "width": "auto",
          "zIndex": "auto",
        }
      }
    >
      <span
        className="normal grayBlue fontSizeSm"
      >
        18th October at 12:30
      </span>
    </div>
  </div>
  <div
    className="block square inherit"
    style={
      Object {
        "height": "auto",
        "margin": 0,
        "padding": 0,
        "width": "auto",
        "zIndex": "auto",
      }
    }
  >
    <span
      className="normal black"
    >
      <a
        href="/users/beautifulpanda108"
        onClick={[Function]}
      >
        <span
          className="normal spanishBlue inline"
        >
          @beautifulpanda108
           
        </span>
      </a>
      <span
        className="normal black inline"
      >
        let's
         
      </span>
      <span
        className="normal black inline"
      >
        get
         
      </span>
      <span
        className="normal black inline"
      >
        coffee
         
      </span>
      <span
        className="normal black inline"
      >
        one
         
      </span>
      <span
        className="normal black inline"
      >
        these
         
      </span>
      <span
        className="normal black inline"
      >
        days
         
      </span>
      <span
        className="normal black inline"
      >
        ☕
         
      </span>
    </span>
  </div>
  <div
    className="block square inherit"
    style={
      Object {
        "height": "auto",
        "margin": 0,
        "padding": 0,
        "width": "auto",
        "zIndex": "auto",
      }
    }
  />
</div>
`;
